### PR TITLE
[libraries] Skip test suites crashing on CI for iOS/tvOS/MacCatalyst

### PR DESCRIPTION
**Pull request by mdh1418**
Numerous test suites have been failing for iOS/tvOS/MacCatalyst consistently on CI without useful logs as to why. Moreover, some of these suites pass locally. 

This PR looks to reduce the failures on CI by skipping the problematic suites
Skips test suites logged in https://github.com/dotnet/runtime/issues/53624

ActiveIssues
https://github.com/dotnet/runtime/issues/58440
https://github.com/dotnet/runtime/issues/58418
https://github.com/dotnet/runtime/issues/58367
https://github.com/dotnet/runtime/issues/58584

### CODE REVIEW

**Comment by mdh1418** [comment]
**8/27 CI Completion**
---------
### Crashed Suites
Android x64 - NONE

Android x86 - NONE

iOSSim arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests
- XsltCompiler.Tests

iOSSim x64
- Microsoft.CSharp.Tests
- System.Diagnostics.Tracing.Tests
- System.Dynamic.Runtime.Tests
- System.Linq.Expressions.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.SourceGeneration.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst x64
- Microsoft.CSharp.Tests
- System.Diagnostics.Tracing.Tests
- System.Dynamic.Runtime.Tests
- System.Linq.Expressions.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.SourceGeneration.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

tvOSSim x64
- Microsoft.CSharp.Tests
- System.Diagnostics.Tracing.Tests
- System.Dynamic.Runtime.Tests
- System.Linq.Expressions.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.SourceGeneration.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

-------
### Test Failures
Android x64 - NONE

Android x86 - NONE

iOSSim arm64
System.IO.MemoryMappedFiles.Tests - 2
System.Security.Cryptography.OpenSsl.Tests - 667
System.IO.Pipes.Tests - 71

iOSSim x64
System.Security.Cryptography.OpenSsl.Tests - 667
System.IO.Pipes.Tests - 71

MacCatalyst arm64
System.Security.Cryptography.OpenSsl.Tests - 667

MacCatalyst x64
System.Security.Cryptography.OpenSsl.Tests - 667

tvOSSim x64
System.Security.Cryptography.OpenSsl.Tests - 667
System.IO.Pipes.Tests - 71

**Comment by mdh1418** [comment]
**8/30 CI Completion**

After having skipped the failing tests in 8/27 run and adopting the Android trampoline settings.
---------
### Crashed Suites
Android x64 - NONE

Android x86 - NONE

iOSSim arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

iOSSim x64
- System.Diagnostics.Tracing.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst x64
- System.Dynamic.Runtime.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

tvOSSim x64
- System.Dynamic.Runtime.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

-------
### Test Failures
Android x64
System.Security.Cryptography.X509Certificates.Tests - 1

Android x86
System.Security.Cryptography.X509Certificates.Tests - 1

iOSSim arm64 - NONE

iOSSim x64 - NONE

MacCatalyst arm64
System.IO.FileSystem.Watcher.Tests - 1

MacCatalyst x64 - NONE

tvOSSim x64 - NONE

**Comment by mdh1418** [comment]
**8/30 2nd CI Completion**

After having skipped the failing tests in 8/30 run and doubling the Android trampoline settings. It seems like trampolines might not be the problem here, so will skip these test suites instead.
---------
### Crashed Suites
Android x64 - NONE

Android x86 - NONE

iOSSim arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

iOSSim x64
- System.Dynamic.Runtime.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.SourceGeneration.Unit.Tests
- System.Threading.Channels.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst x64
- System.Dynamic.Runtime.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

tvOSSim x64
- System.Dynamic.Runtime.Tests
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

-------
### Test Failures
Android x64 - NONE

Android x86 - NON

iOSSim arm64
System.Tests.TimeZoneInfoTests.IsIanaIdTest
     should not be IANA Id.\nExpected: False\nActual:   True

iOSSim x64 
System.Tests.TimeZoneInfoTests.IsIanaIdTest
     should not be IANA Id.\nExpected: False\nActual:   True

MacCatalyst arm64
System.Tests.TimeZoneInfoTests.IsIanaIdTest
     should not be IANA Id.\nExpected: False\nActual:   True

MacCatalyst x64 
System.Tests.TimeZoneInfoTests.IsIanaIdTest
     should not be IANA Id.\nExpected: False\nActual:   True

tvOSSim x64 
System.Tests.TimeZoneInfoTests.IsIanaIdTest
     should not be IANA Id.\nExpected: False\nActual:   True

File: src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs [hunk 1]
```diff
@@ -97,6 +97,11 @@ public void ValidAccessLevelCombinations(MemoryMappedFileAccess mapAccess, Memor
                         // Containers and OSX with SIP enabled do not have execute permissions by default.
                         throw new SkipTestException("Insufficient execute permission.");
                     }
+                    else if ((OperatingSystem.IsIOS() && PlatformDetection.IsArm64Process) &&
```
Comment (akoeplinger) [inline src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs:100]: I think this isn't specific to arm64, it should happen on all iOS devices and simulators as well as on tvOS.

We can probably just add `|| OperatingSystem.IsIOS() || OperatingSystem.IsTvOS()` to the if above (and update the comment).

File: src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedViewStream.Tests.cs [hunk 2]
```diff
@@ -97,6 +97,11 @@ public void ValidAccessLevelCombinations(MemoryMappedFileAccess mapAccess, Memor
                         // Containers and OSX with SIP enabled do not have execute permissions by default.
                         throw new SkipTestException("Insufficient execute permission.");
                     }
+                    else if ((OperatingSystem.IsIOS() && PlatformDetection.IsArm64Process) &&
```
Comment (akoeplinger) [inline src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedViewStream.Tests.cs:100]: same here

File: src/libraries/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs [hunk 3]
```diff
@@ -886,6 +886,7 @@ public static void X509CertificateCollectionCopyTo()
 
         [Fact]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/57506", typeof(PlatformDetection), nameof(PlatformDetection.IsMonoRuntime), nameof(PlatformDetection.IsMariner))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/58416", TestPlatforms.Android)]
```
Comment (akoeplinger) [inline src/libraries/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs:889]: please wait until https://github.com/dotnet/runtime/pull/58441 goes in and revert this then.

**Comment by mdh1418** [comment]
**9/1 CI Completion**

After having 8x the trampoline settings Android has for iOS/MacCatalyst/tvOS which was already above the defaults, its hard to see what trampoline bump would be reasonable for iOS/MacCatalyst/tvOS. Its not clear what exactly is limiting from the trampolines that causes these suite failures, so this PR will now skip the test suites.
---------
### Crashed Suites
Android x64 - NONE

Android x86 - NONE

iOSSim arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

iOSSim x64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst arm64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

MacCatalyst x64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

tvOSSim x64
- System.Net.Quic.Functional.Tests
- System.Net.Requests.Tests
- System.Net.Security.Tests
- System.Text.Json.Tests
- System.Threading.Tasks.Dataflow.Tests
- System.Threading.Tasks.Tests

-------
### Test Failures
Android x64 - NONE

Android x86 - NON

iOSSim arm64 - NONE

iOSSim x64  - NONE

MacCatalyst arm64 - NONE

MacCatalyst x64  - NONE

tvOSSim x64 - NONE

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**